core/src/main/java/org/infinispan/interceptors/IsMarshallableInterceptor.java

```diff
@@ -95,15 +95,20 @@ public Object visitGetAllCommand(InvocationContext ctx, GetAllCommand command) t
 
    @Override
    public Object visitLockControlCommand(TxInvocationContext ctx, LockControlCommand command) throws Throwable {
-      if (isStoreAsBinary() || isClusterInvocation(ctx, command))
-         checkMarshallable(command.getKeys());
+      if (isStoreAsBinary() || isClusterInvocation(ctx, command)) {
```

Comment:
Yeah, I should have kept it in a separate commit... it's not about allocations, the problem was that `checkMarshallable` was checking if the collection was serializable, instead of checking each key.